forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -67,31 +66,40 @@ pub struct SnapshotStore {
 impl SnapshotStore {
     pub fn new(config: &Config, snapshot_dir: &PathBuf) -> SnapshotStore {
         let mut snapshots = Vec::new();
+        let pattern = Regex::new(
+            r"^([^_]+?)_snapshot_(?P<network>[^_]+?)_(?P<date>\d{4}-\d{2}-\d{2})_height_(?P<height>\d+).car(.tmp|.aria2)?$",
+        ).unwrap();
         if let Ok(dir) = std::fs::read_dir(snapshot_dir) {
             dir.flatten()
-            .map(|entry| entry.path())
-            .filter(|p| is_car_or_tmp(p))
-            .for_each(|path|
-                if let Some(filename) = path.file_name().and_then(|n| n.to_str()) {
-                    let pattern = Regex::new(
-                        r"^([^_]+?)_snapshot_(?P<network>[^_]+?)_(?P<date>\d{4}-\d{2}-\d{2})_height_(?P<height>\d+).car(.tmp|.aria2)?$",
-                    ).unwrap();
-                    if let Some(captures) = pattern.captures(filename) {
-                        let network: String = captures.name("network").unwrap().as_str().into();
-                        if network == config.chain.name {
-                            let date = Date::parse(captures.name("date").unwrap().as_str(), &Iso8601::DEFAULT).unwrap();
-                            let height = captures.name("height").unwrap().as_str().parse::<i64>().unwrap();
-                            let snapshot = SnapshotInfo {
-                                network,
-                                date,
-                                height,
-                                path,
-                            };
-                            snapshots.push(snapshot);
+                .map(|entry| entry.path())
+                .filter(|p| is_car_or_tmp(p))
```

Comment:
maybe `filter_map` instead?

Reply:
@creativcoder not sure if `filter_map` will be right fit here
we will have to do something like this to use it
```
.filter_map(|entry| {
     if is_car_or_tmp(&entry.path()) {
        Some(entry.path())
     } else {
        None
     }
})
```